File: src/api/routes/sellers.py
"""关注卖家 API"""
from fastapi import APIRouter, HTTPException, Query
from pydantic import BaseModel

from src.services import seller_service

router = APIRouter(prefix="/api/sellers", tags=["sellers"])


class FollowRequest(BaseModel):
    seller_id: str
    seller_name: str = ""
    avatar_url: str = ""
    city: str = ""
    cron: str = "0 */6 * * *"
    note: str = ""


class CronUpdate(BaseModel):
    cron: str


@router.get("")
async def list_sellers():
    return {"items": seller_service.list_followed_sellers()}


@router.post("")
async def follow_seller_api(body: FollowRequest):
    return seller_service.follow_seller(
        body.seller_id,
        seller_name=body.seller_name,
        avatar_url=body.avatar_url,
        city=body.city,
        cron=body.cron,
        note=body.note,
    )


@router.delete("/{seller_id}")
async def unfollow_seller_api(seller_id: str):
    if not seller_service.unfollow_seller(seller_id):
        raise HTTPException(status_code=404, detail="未关注该卖家")
    return {"message": "已取消关注"}


@router.get("/{seller_id}")
async def get_seller_api(seller_id: str, sort: str = Query("want_cnt")):
    seller = seller_service.get_followed_seller(seller_id)
    if not seller:
        raise HTTPException(status_code=404, detail="未关注该卖家")
    items = seller_service.get_seller_items(seller_id, sort_by=sort)
    return {"seller": seller, "items": items}


@router.post("/{seller_id}/refresh")
async def refresh_seller_api(seller_id: str):
    return await seller_service.refresh_seller_items(seller_id)


@router.post("/refresh-all")
async def refresh_all_sellers_api():
    return {"results": await seller_service.refresh_all_sellers()}


@router.put("/{seller_id}/cron")
async def update_cron_api(seller_id: str, body: CronUpdate):
    if not seller_service.update_seller_cron(seller_id, body.cron):
        raise HTTPException(status_code=404, detail="未关注该卖家")
    return {"message": "cron 已更新", "cron": body.cron}

File: web-ui/src/api/sellers.ts
import { http } from '@/lib/http'

export interface FollowedSeller {
  seller_id: string
  unique_name?: string
  portrait_url?: string
  city?: string
  signature?: string
  item_count?: number
  has_sold_num?: number
  new_good_ratio_rate?: string
  user_reg_day?: number
  xianyu_summary?: string
  remark_good_cnt?: number
  remark_default_cnt?: number
  remark_bad_cnt?: number
  cron: string
  follow_at: string
  last_fetch_at?: string
  last_fetch_status: string
  last_fetch_error?: string
  note?: string
}

export interface SellerItem {
  item_id: string
  title: string
  price: string
  pic_url: string
  status: string
  want_cnt?: number
  browse_cnt?: number
  collect_cnt?: number
  sold_cnt?: number
  fetched_at: string
}

export async function listSellers(): Promise<{ items: FollowedSeller[] }> {
  return await http('/api/sellers')
}

export async function followSeller(payload: {
  seller_id: string
  seller_name?: string
  avatar_url?: string
  city?: string
  cron?: string
  note?: string
}): Promise<{ seller_id: string; followed: boolean }> {
  return await http('/api/sellers', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  })
}

export async function unfollowSeller(sellerId: string): Promise<{ message: string }> {
  return await http(`/api/sellers/${sellerId}`, { method: 'DELETE' })
}

export async function getSeller(sellerId: string, sort = 'want_cnt'): Promise<{ seller: FollowedSeller; items: SellerItem[] }> {
  return await http(`/api/sellers/${sellerId}?sort=${sort}`)
}

export async function refreshSeller(sellerId: string): Promise<{ seller_id: string; status: string; items_count?: number; error?: string }> {
  return await http(`/api/sellers/${sellerId}/refresh`, { method: 'POST' })
}

export async function refreshAllSellers(): Promise<{ results: any[] }> {
  return await http('/api/sellers/refresh-all', { method: 'POST' })
}

export async function updateSellerCron(sellerId: string, cron: string): Promise<{ message: string; cron: string }> {
  return await http(`/api/sellers/${sellerId}/cron`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ cron }),
  })
}
